src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2495,7 +2495,7 @@ protected virtual TypeSymbol GetCurrentReturnType(out RefKind refKind)
 
                 TypeSymbolWithAnnotations returnType = symbol.ReturnType;
 
-                if (returnType.IsNull || (object)returnType.TypeSymbol == LambdaSymbol.ReturnTypeIsBeingInferred)
+                if (returnType.IsDefault || (object)returnType.TypeSymbol == LambdaSymbol.ReturnTypeIsBeingInferred)
```

Comment:
>IsDefault [](start = 31, length = 9)

HasType?

Reply:
The test is unnecessary. Removing it.

---
In reply to: [260055037](https://github.com/dotnet/roslyn/pull/33648#discussion_r260055037) [](ancestors = 260055037)